design/proposals/load_policy/load_policy_validation.md

```diff
@@ -0,0 +1,285 @@
+related the JWT authenticator
+   
+7. new ValidateJwtHostConstraints command class - is responsible to to perform the validation logic as written in the requirements,
+for all annotations have values.
+   
+    Note: we will refactor the current code of the authentication flow to use this new command class as well 
+   
+
+   
+
+## Performance
+* We will have a performance degradation in policy load action, 
+  we will know the degradation percent after we will implements the performance tests.
+
+## Backwards Compatibility
+* Host with JWT annotations that were loaded successfully in the current Conjur version 
+  will probably failed in load policy since they need modifications to answer the new constraints. 
+
+## Affected Components
+* All components, the load policy is a main feature
+
+## Test Plan
+Note: this is an high level test plan, we will need to extend and challenge it as this feature will enter into
+development stage
+
+#### Regression 
+Valid policy without Hosts entities, policy validation is not activated, look for new logs, 200 OK
+
+Invalid policy with syntax error, policy validation is not activated, Same error as previous version
+
+Load policy without permissions,  policy validation is not activated, Same error as previous version
+
+Load policy with empty value,  policy validation is not activated, Same error as previous version
+
+Load policy with just policy id,  policy validation is not activated, Same error as previous version
+
+#### Functionality 
+Load policy with 1 host with JWT annotations, policy validation is  activated with expected output
+* Write both success and failure scenarios 
+
+Load nested policies with 5 hosts with JWT annotations, policy validation is  activated with expected output
+* Write both success and failure scenarios 
+
+Load hosts with and without JWT annotations in same policy,  policy validation is  activated with expected output
+* Write both success and failure scenarios 
+
+#### Performance
+Run performance tests of load policy before and after the feature with and without JWT hosts 
+
+#### Load
+Load policy with 100 host with JWT annotations, policy validation is  activated with expected output
+
+### Out of Scope
+* This design do not handled the upgrade process and how to migrate invalid hosts from 
+old version to new version.
+
+## Documentation
+* We will need to document the UX for all kinds of put/post/patch in case of changing/not changing
+the DB, since the validation occurs only on the DB final state.
+  So there are possible use cases were the policy input will be invalid 
+  (For example: empty annotations), but the validation will passed successfully since the DB may not changed.
+
+## Security
+TBD
+
+## Infrastructure
+None.
+
+## Audit
+* No need for new audit records 
+
+## Open Questions
+None.
+
+## Delivery Plan
+
+### 1st plan - 1 milestone Total EE 45
```

Comment:
what is the difference between 1st and 2nd plan?

Reply:
1st plan - the whole feature 1 shot 
2nd plan - the 1st plan split into 2 milestones